platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/parser/PomReader.java

```diff
@@ -728,6 +739,10 @@ private Map<String, String> parseProperties(Element parentElement) {
         return pomProperties;
     }
 
+    private static boolean hasUnresolvedProperty(String val) {
+        // Variable substitution uses $
+        return val.contains("$");
```

Comment:
Is this necessarily true? Can you have a `$` that is not a pattern? Can we expand this to a regex that checks for `$\{.*\}`?